manticore/core/state.py

```diff
@@ -301,6 +301,23 @@ def migrate_expression(self, expression):
     def is_feasible(self):
         return self.can_be_true(True)
 
+    # alternate names:
+    # state.check
+    # state.verify
+    # state.ensure
+    # ...
+    def assert_(self, expression, testcase_message, testcase_name='Assertion Failed'):
```

Comment:
assert, verify, ensure kind of suggest to me something is done to the state, no?

This is not modifying the current state. We may even put this as a helper?
manticore.generate_testcase_if_fail(state, constraint, message)

(or if it breaks lots of scripts carrying the manticore back link in all states as we discussed)